api/api/models_api.go

```diff
@@ -99,3 +102,119 @@ func (c *ModelsController) GetModel(r *http.Request, vars map[string]string, bod
 
 	return Ok(model)
 }
+
+// DeleteModel delete a model given a project and model ID
+func (c *ModelsController) DeleteModel(r *http.Request, vars map[string]string, body interface{}) *Response {
+	ctx := r.Context()
+
+	projectID, err := models.ParseID(vars["project_id"])
+	if err != nil {
+		return BadRequest("Unable to parse project_id")
+	}
+	modelID, err := models.ParseID(vars["model_id"])
+	if err != nil {
+		return BadRequest("Unable to parse model_id")
+	}
+
+	_, err = c.ProjectsService.GetByID(ctx, int32(projectID))
+	if err != nil {
+		return NotFound(err.Error())
+	}
+
+	model, err := c.ModelsService.FindByID(ctx, modelID)
+	if err != nil {
+		if gorm.IsRecordNotFoundError(err) {
+			return NotFound(fmt.Sprintf("Model is not found: %s", err.Error()))
+		}
+		return InternalServerError(err.Error())
+	}
+
+	// get all model version for this model
+	versions, _, err := c.VersionsService.ListVersions(ctx, modelID, c.MonitoringConfig, service.VersionQuery{})
+	if err != nil {
+		return InternalServerError(err.Error())
+	}
+
+	var inactiveJobsInModel [][]*models.PredictionJob
+	var inactiveEndpointsInModel [][]*models.VersionEndpoint
+
+	// iterate for every version, check the active prediction job / active endpoint
+	for _, version := range versions {
+		if model.Type == "pyfunc_v2" {
+			// check active prediction job
+			// if there are any active prediction job using the model version, deletion of the model version are prohibited
+			jobs, response := c.VersionsController.getInactivePredictionJobsForDeletion(ctx, model, version)
```

Comment:
yes, it always return error, let me change it to `errResponse`, the reason that i don't use `error`, is because the return type is different, with the function returnting struct `response` immediately, so thats why i don't name it `error`